### ISSUE TITLE
[Feature request] Raycast integration

### ISSUE THREAD

**Issue by howarddo2208** [omnivore-app/omnivore#1578]
Hi, I'm a Mac user and use [Raycast](https://www.raycast.com/) instead of Spotlight because it offers more functionalities and customization

I used raindrop as a read later app before, and it comes with Raycast extension that I can give my API key, then I can search pages right inside the Raycast prompt without needing to navigate through the app.

It would be nice if omnivore got this integration.

**Comment by jacksonh** [comment]
Thanks @howarddo2208 is search the main functionality you'd like, or is there other functions like saving?

**Comment by howarddo2208** [comment]
https://www.raycast.com/lardissone/raindrop-io/commands

It only has 2 commands: latest bookmarks and search, I think saving is not necessary as I would use the Chrome extension instead

**Comment by jacksonh** [comment]
Thanks, that makes sense. It looks quite easy to implement but probably be a week or two before we can get it on our roadmap.

**Comment by howarddo2208** [comment]
Thanks!

**Comment by jacksonh** [comment]
Yeah cool idea, thank you.

**Comment by jacksonh** [comment]
@marcjulianschwarz i did a tiny amount of work but wasn't able to complete. Would love it if you want to keep going on it.

**Comment by marcjulianschwarz** [comment]
@jacksonh I saw that the obsidian integration has its repo in the omnivore-app organization. Would you consider it to be helpful to have the raycast extension in here as well such that the omnivore open source community can easily contribute?

Of course I could first work on a prototype and then when it’s somewhat stable we can talk about adding it here. What do you think?

**Comment by jacksonh** [comment]
Yeah definitely would be better I think, just easier for people to find. I think the easiest way to do this is if you put together a prototype, then i can create a new repo here, and give you access to it

**Comment by marcjulianschwarz** [comment]
Sounds great, will let you know when I have something ready to show :)

**Comment by jacksonh** [comment]
Yeah and please feel free to jump into discord and ask about any API questions

**Comment by aurelticot** [comment]
> Is someone already working on this? I have created a prototype extension which already works quite well for viewing articles and highlights and notes. Shouldn’t be too hard to implement other commands like saving too.
> 
> I’d love to work more on this, so let me know whether I could contribute this integration. (btw, I created the [Raycast extension for obsidian](https://github.com/KevinBatdorf/obsidian-raycast)). This one should be a lot easier as the omnivore API is already really powerful.

Ah 😱, I started working on it as well, about two weeks ago, seeing this ticket not being addressed yet. I should have left a comment 🤦‍♂️.

I just pushed the code to GitHub to present it to @jacksonh. You can find it on [aurelticot/raycast-omnivore](https://github.com/aurelticot/raycast-omnivore)

@marcjulianschwarz I just saw your updates on Discord, the AI use is pretty cool.

Anyway, I will hold off from publishing my extension but happy to contribute if it comes in the main Omnivore repo.

Cheers

**Comment by marcjulianschwarz** [comment]
@aurelticot  I took a look at your code, and it seems much more refined compared to the code in my prototype. I'm all for merging your code into the main omnivore repo. Additionally, I'd be happy to contribute some of my functions to complement what you've done. 🎉

**Comment by aurelticot** [comment]
@marcjulianschwarz sure let's collaborate on it.

To be honest I'm not familiar with publishing a Raycast extension. This Omnivore one was my first 
I read their documentation, apparently it's forking their extensions repo, .
Will it be possible to publish if the extension is within a monorepo?

**Comment by aurelticot** [comment]
@marcjulianschwarz sure let's collaborate on it.

To be honest I'm not familiar with publishing a Raycast extension. This Omnivore one was my first hands on developing one.
I read their documentation, apparently it's automated to fork the Raycast Extensions repo, put the extension code inside, and create a PR for their review.

Will it work if the extension is within a monorepo (the Omnivore one)?
And is it convenient to continue maintaining the extension in its own repo/monorepo rather than directly in the Raycast Extensions one?

**Comment by marcjulianschwarz** [comment]
In my opinion it would be beneficial to have the source code in the omnivore organization so that potential contributors can easily find it. 

When I was working on the Obsidian extension, I maintained my own repo and always copied over all files (basically a release) to a fork of the raycast extensions repo, then created a pull request, then deleted the fork again. 
I think this release workflow is not that much of an effort. 

The only disadvantage that could be a bit annoying is the fact that we would have two repos where issues will be created. For the obsidian extension I used a GitHub project where I could capture issues from both sources. Also this project acted as a roadmap, so the community could always see what was being worked on.

**Comment by aurelticot** [comment]
> In my opinion it would be beneficial to have the source code in the omnivore organization so that potential contributors can easily find it.

Agreed

> When I was working on the Obsidian extension, I maintained my own repo and always copied over all files (basically a release) to a fork of the raycast extensions repo, then created a pull request, then deleted the fork again. I think this release workflow is not that much of an effort.
> The only disadvantage that could be a bit annoying is the fact that we would have two repos where issues will be created...

Seems reasonable for now

How do we want to proceed from here @marcjulianschwarz @jacksonh? Are we ok to go forward? Maybe worth discussing the detail son Discord?

**Comment by marcjulianschwarz** [comment]
@aurelticot I just installed your extension locally and it looks and feels great. I think it is definitely ready to be published to the Raycast store. 

I'd love to contribute some of the [AI features (YouTube)](https://youtu.be/-xj5pCccDwk) that I showed in my initial version. However I'd still wait for the first release and then add features when everything is kind of stable for the users.
